homeassistant/components/fronius/quality_scale.yaml

```diff
@@ -0,0 +1,79 @@
+rules:
+  # Bronze
+  action-setup:
+    status: exempt
+    comment: |
+      This integration does not provide additional actions.
+  appropriate-polling: done
+  brands: done
+  common-modules: done
+  config-flow-test-coverage: done
+  config-flow: done
+  dependency-transparency: done
+  docs-actions:
+    status: exempt
+    comment: |
+      This integration does not provide additional actions.
+  docs-high-level-description: done
+  docs-installation-instructions: done
+  docs-removal-instructions: done
+  entity-event-setup:
+    status: exempt
+    comment: |
+      This integration does not subscribe to events.
+  entity-unique-id: done
+  has-entity-name: done
+  runtime-data: done
+  test-before-configure: done
+  test-before-setup: done
+  unique-config-entry: done
+
+  # Silver
+  action-exceptions:
+    status: exempt
+    comment: |
+      This integration does not provide additional actions.
+  config-entry-unloading: done
+  docs-configuration-parameters:
+    status: exempt
+    comment: |
+      This integration does not provide configuration options.
+  docs-installation-parameters: done
+  entity-unavailable: done
+  integration-owner: done
+  log-when-unavailable: done
+  parallel-updates: done
+  reauthentication-flow:
+    status: exempt
+    comment: |
+      This integration doesn't require authentication.
+  test-coverage: done
+  # Gold
+  devices: done
+  diagnostics: done
+  discovery-update-info: done
+  discovery: done
+  docs-data-update: done
+  docs-examples: done
+  docs-known-limitations: done
+  docs-supported-devices: done
+  docs-supported-functions: done
+  docs-troubleshooting: done
+  docs-use-cases: done
+  dynamic-devices: done
+  entity-category: done
+  entity-device-class: done
+  entity-disabled-by-default: done
+  entity-translations: done
+  exception-translations: done
+  icon-translations: done
+  reconfiguration-flow: done
+  repair-issues:
+    status: exempt
+    comment: |
+      This integration doesn't have any known user-repairable issues.
+  stale-devices: done
```

Comment:
you're adding devices, but where are you removing them?

Reply:
It is manual removal since solar inverters might not be online during night (maybe even couple days in winter) and come back when the sun shines.

Reply:
let's set it to exempt then?

Reply:
This is explicitly mentioned as valid way in https://developers.home-assistant.io/docs/core/integration-quality-scale/rules/stale-devices
> We should only remove devices that we are sure are no longer available. If you can't be sure if a device is still available, be sure to implement async_remove_config_entry_device. This allows the user to delete the device from the device registry manually.

Reply:
ah right, you have `async_remove_config_entry_device`, fine by me then

Reply:
If the solar inverter is offline, does it still provide the full data? Or does it look like the extra devices or entities are not there when that happens?

Reply:
When the inverter is offline it's data might just be omitted by any API response. This depends on an inverter setting - it can be set to power down when no PV energy or not. A primary inverter with a meter will probably be set not to power down, but additional inverters, or such without smart meter, are usually configured to power off.